### ISSUE TITLE
False positive when using "/" for positional arguments (too-many-positional-arguments)

### ISSUE THREAD

**Issue by DeveloperAnonymous** [pylint-dev/pylint#10122]
### Bug description
R0917: Too many positional arguments (7/5) (too-many-positional-arguments)
Pylint has trouble detecting positional arguments when using "/". In there, only a, b and c are positional while d, e and f are keyword parameters.

This behavior did not happen with pylint 3.0.3, astroid 3.0.3

```python
class Foo:
    def __init__(self, a, b, c, /, d=None, e=None, f=None):
        pass
```

### Configuration

```ini
[MASTER]
# Maximum number of characters on a single line
max-line-length=120
min-public-methods=0

[MESSAGES CONTROL]
# Disabled the message "C0114: Missing module docstring"
# Disabled the message "W0511: fixme"
# Disabled the message "R0913: Too many arguments"
# Disabled the message "W0212: Access to a protected member _ of a client class"
disable=C0114, W0511, R0913, W0212

[FORMAT]
# Disabled the message "W0603: Using the global statement"
disable=W0603
```

### Command used

```shell
pylint $(git ls-files '*.py')
```

### Pylint output

```python
(I'm trimming the output because its part of a larger project)

R0917: Too many positional arguments (7/5) (too-many-positional-arguments)
```

### Expected behavior

The expected behavior would be to accept that definition as it really only has 3 positional arguments, not 7

### Pylint version

```shell
pylint 3.3.2
astroid 3.3.6
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22.04.3 LTS, WSL2

### Additional dependencies

```python

```

**Comment by mbyrnepr2** [comment]
There's 3 positional-only parameters(a, b, c) and 3 positional-or-keyword parameters(d, e, f) there.
(I would say that there is a false positive because `self` appears to be counted here).

The [check](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/too-many-positional-arguments.html) is encouraging you to add `*` so that calls to this function explicitly require keywords:

```python
class Foo:
    def __init__(self, a, b, c, /, *, d=None, e=None, f=None):
        pass
```

**Comment by jacobtylerwalls** [comment]
If your project disables too-many-arguments, you probably want to disable this new message also.
